File: leetsync/1037-minimum-number-of-k-consecutive-bit-flips/minimum-number-of-k-consecutive-bit-flips.go
func minKBitFlips(nums []int, k int) int {
	n := len(nums)
	isFlipped := make([]bool, n)
	totalFlips := 0
	flipCountFromPastForCurrentI := 0
	for i := 0; i < n; i++ {
		if i >= k && isFlipped[i-k] {
			flipCountFromPastForCurrentI--
		}
		if flipCountFromPastForCurrentI%2 == nums[i] {
			if (i + k) > n {
				return -1
			}
			totalFlips++
			isFlipped[i] = true
			flipCountFromPastForCurrentI++
		}
	}
	return totalFlips
}

File: leetsync/1037-minimum-number-of-k-consecutive-bit-flips/minimum-number-of-k-consecutive-bit-flips.java
class Solution {
    public int minKBitFlips(int[] nums, int k) {
        int n = nums.length;
        boolean[] isFlipped = new boolean[n];
        int totalFlips = 0;
        int flipCountFromPastForCurrentI = 0;
        for (int i = 0; i < n; i++) {
            if (i >= k && isFlipped[i - k]) {
                flipCountFromPastForCurrentI--;
            }
            if (flipCountFromPastForCurrentI % 2 == nums[i]) {
                if (i + k > n) {
                    return -1;
                }
                totalFlips++;
                isFlipped[i] = true;
                flipCountFromPastForCurrentI++;
            }
        }
        return totalFlips;
    }
}